interfaces/builtin/content_test.go

```diff
@@ -376,3 +376,150 @@ slots:
 func (s *ContentSuite) TestInterfaces(c *C) {
 	c.Check(builtin.Interfaces(), testutil.DeepContains, s.iface)
 }
+
+func (s *ContentSuite) TestModernContentInterface(c *C) {
+	plug := MockPlug(c, `name: consumer
+plugs:
+ content:
+  target: $SNAP_COMMON/import
+apps:
+ app:
+  command: foo
+`, &snap.SideInfo{Revision: snap.R(1)}, "content")
+
+	slot := MockSlot(c, `name: producer
+slots:
+ content:
+  source:
+    read:
+     - $SNAP_COMMON/read-common
+     - $SNAP_DATA/read-data
+     - $SNAP/read-snap
+    write:
+     - $SNAP_COMMON/write-common
+     - $SNAP_DATA/write-data
```

Comment:
What happens if both read and write have same entries?

Reply:
This is pretty interesting, we'll get a clash and those get de-conflicted. I'll add a test that shows what would happen.